src/sources/journald.rs

```diff
@@ -552,31 +598,76 @@ impl Drop for RunningJournalctl {
     }
 }
 
-fn create_event(record: Record, batch: &Option<BatchNotifier>) -> LogEvent {
-    let mut log = LogEvent::from_iter(record).with_batch_notifier_option(batch);
+fn create_event(
+    record: Record,
+    batch: &Option<BatchNotifier>,
+    log_namespace: LogNamespace,
+) -> LogEvent {
+    let mut log = LogEvent::default().with_batch_notifier_option(batch);
+
+    // Add the fields from the Record to the log event.
+    record.iter().for_each(|(key, value)| {
+        log_namespace.insert_source_metadata(
+            JournaldConfig::NAME,
+            &mut log,
+            Some(LegacyKey::Overwrite(path!(key))),
+            path!(key),
+            value.as_str(),
+        );
+    });
 
     // Convert some journald-specific field names into Vector standard ones.
     if let Some(message) = log.remove(MESSAGE) {
         log.insert(log_schema().message_key(), message);
     }
```

Comment:
@StephenWakely  are you suggesting that we change this to something like:

```suggestion
    // Convert some journald-specific field names into Vector standard ones.
    if let Some(message) = log.remove(MESSAGE) {
        
        log_namespace.insert_vector_metadata(
            &mut log,
            log_schema().message_key(),
            path("message"),
            message,
    }
```

Reply:
Or, would the message just be inserted at the root?

Reply:
> This is going to be really annoying of me and I apologise in advance. Do we actually want to make message a top level field for the Vector namespace if it exists in the journald message?

Not annoying! I'd rather get it right.

I had intentionally left that untouched as I hadn't seen a similar case to it but now that you point this out, I could see it being more correct this way.

Reply:
I think something like:

```rust
     let log = match log_namespace {
        LogNamespace::Legacy => {
            let mut log = LogEvent::from_iter(record);
            if let Some(message) = log.remove(MESSAGE) {
                log.insert(log_schema().message_key(), message);
            }      
      
            log
        }
        LogNamespace::Vector => {
            let log = LogEvent::from(Value::Bytes(Bytes::from(record.remove(MESSAGE).unwrap_or(""))));

            for (key, val) in record {
                log.metadata_mut()
                  .value_mut()
                  .insert(path!("journald").concat(key), val);
            }
            
            log
        }
    };
```

Reply:
I voiced my preference in slack regarding empty string vs null - but this is generally 👍 however, the `for (key, val) in record` would need to be inserted into an object in the source metadata, as to avoid key collisions.

Reply:
> however, the for (key, val) in record would need to be inserted into an object in the source metadata, as to avoid key collisions.

Just wanted to point out this is my current code on branch that is doing that, and according to your comment, it sounds like you're suggesting a collection to contain it within like `"record"` instread ?
This is something I'm pretty sure we discussed in a prior thread on this PR already :sob: 


```Rust
   // Add the fields from the Record to the log event.
    record.iter().for_each(|(key, value)| {
        log_namespace.insert_source_metadata(
            JournaldConfig::NAME,
            &mut log,
            Some(LegacyKey::Overwrite(path!(key))),
            path!(key),
            value.as_str(),
        );
    });
 ```

Reply:
Took a swing in https://github.com/vectordotdev/vector/pull/15326/commits/e15958c496801642494d1ad1a3815e3ce76bfdee